### ISSUE TITLE
quote-props should (optionally) not complain about inconsistent quotes if quotes are required on some

### ISSUE THREAD

**Issue by jamestalmage** [eslint/eslint#3759]
Assuming your preference is not to use quotes around property names, the following presents a problem:

```js
var x = {
  foo: 'bar',
  baz: 'quz',
  // ... many more
  default: 'or some other reserved word',
  'required-because': 'it has a dash',
  // ...
  y: 'z'
};
```

Rules introduced recently (whatever `xo@0.8` depends, ) now fail because `default` is a reserved word. Adding quotes around `default` causes eslint to complain about the inconsistency.

It seems to me that "using quotes only when it's required" qualifies as "consistent".

I did poke around in the source briefly, and I don't think the behavior I expect is just a case of misconfiguration, but I'm eager to be enlightened if that is the case.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by platinumazure** [comment]
Can you include your configuration (`.eslintrc`)?

**Comment by jamestalmage** [comment]
sorry,
don't have one of those.
I am using `xo`.

I poked around [their source](https://github.com/sindresorhus/xo/blob/master/index.js#L24) a little, and their is a reference to

```js
baseConfig: {
  extends: 'xo'
}
```

I'm sure that is referencing some default config somewhere, but couldn't figure it out.

**Comment by sindresorhus** [comment]
It's working as intended.

This is the change you're looking for: https://github.com/sindresorhus/eslint-config-xo/commit/cc7d2bd26f0eff756a800f9bc87cc20928485b7b

And here's the rule: http://eslint.org/docs/rules/quote-props#consistent-as-needed

> Rules introduced recently (whatever xo@0.8 depends, ) now fail because default is a reserved word. 

That's incorrect. It complains as you're not consistently quoting the rules. If you quote one, you should quote the others too for consistency.

**Comment by gyandeeps** [comment]
@sindresorhus we can safely close this issue?

**Comment by jamestalmage** [comment]
It definitely was complaining that default was a reserved keyword, even though it was in an object literal.

I am fine with closing this. But another should be opened with regard to the default keyword.
